From 8f1d61e9902cdcb583d4a9053de24b71ebcd3e35 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Tue, 28 Jun 2016 01:14:18 +0800
Subject: [PATCH 532/538] fix: usb: a3700: fix usb crash when COMPHY0 is not
 usb

This patch disables the USB3 in DT node in case the
COMPHY0 is not initialized as USB.
JIRA: SYSTEMSW-2593 Exception stack happens at starting
with RGMII and SGMII 1.

Change-Id: Ie8f0a2f63723a26f4aa1fbbe20e1f9382b6fb3aa
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30743
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db-sfp.dts    | 3 +++
 arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-sfp.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-sfp.dts
index e7827dd..1adfb2a 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db-sfp.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-sfp.dts
@@ -64,6 +64,9 @@
 					full-duplex;
 				};
 			};
+			usb3@58000 {
+				status = "disabled";
+			};
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts
index e74d732..3ad6ddf 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts
@@ -56,6 +56,9 @@
 				phy = <&phy1>;
 				status = "okay";
 			};
+			usb3@58000 {
+				status = "disabled";
+			};
 		};
 	};
 };
-- 
1.9.1

